graphql-server/src/utils/oncotree.ts

```diff
@@ -0,0 +1,112 @@
+import fetch from "node-fetch";
+import NodeCache from "node-cache";
+import { driver } from "../schemas/neo4j";
+import { props } from "./constants";
+
+/**
+ * Source: https://oncotree.mskcc.org/#/home?tab=api
+ */
+export type OncotreeTumorType = {
+  children: Record<string, unknown>;
+  code: string;
+  color: string;
+  externalReferences: Record<string, unknown>;
+  history: string[];
+  level: number;
+  mainType: string;
+  name: string;
+  parent: string;
+  precursors: string[];
+  revocations: string[];
+  tissue: string;
+};
+
+export type CachedOncotreeData =
+  | Pick<OncotreeTumorType, "name" | "mainType">
+  | undefined;
```

Comment:
TypeScript's `Pick<>` lets me "pick" `name` and `mainType` props from `OncotreeTumorType`. This is equivalent to the following:

```ts
export type CachedOncotreeData = {
    name: string;
    mainType: string;
} | undefined;
```

I wanted to avoid overloading the cache with unnecessary data, keeping it light.